Sources/PersistentCollections/Node/_Node+isSubset.swift

```diff
@@ -0,0 +1,71 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension _Node {
+  /// Returns true if `self` contains a subset of the keys in `other`.
+  /// Otherwise, returns false.
+  @inlinable @inline(never)
+  internal func isSubset<Value2>(
+    _ level: _Level,
+    of other: _Node<Key, Value2>
+  ) -> Bool {
+    if self.raw.storage === other.raw.storage { return true }
+    guard self.count <= other.count else { return false }
+
+    if self.isCollisionNode {
+      // Beware, self might be on a compressed path
+      return read {
+        let items = $0.reverseItems
+        let hash = $0.collisionHash
+        return items.indices.allSatisfy {
+          // FIXME: This will repeatedly & unnecessarily call `collisionHash`
+          other.containsKey(level, items[$0].key, hash)
+        }
+      }
+    }
```

Comment:
What was the rationale to extract `_isDisjointCollision` but keep the code inlined for `isSubset`?